pkg/apis/pingcap/v1alpha1/tidbcluster_component.go

```diff
@@ -164,6 +166,17 @@ func (a *componentAccessorImpl) NodeSelector() map[string]string {
 	return sel
 }
 
+func (a *componentAccessorImpl) Labels() map[string]string {
+	l := map[string]string{}
+	for k, v := range a.clusterLabels {
+		l[k] = v
+	}
+	for k, v := range a.ComponentSpec.Labels {
```

Comment:
do we need to check `if a.ComponentSpec != nil` as `Annotations` did?

Reply:
I don't know.. just check in https://github.com/pingcap/tidb-operator/pull/3892/commits/67dc55eee71d7d12d84fd69486bb8461c61bcfc4